ext/openssl/openssl.c

```diff
@@ -4810,10 +4786,18 @@ PHP_FUNCTION(openssl_pkey_derive)
 			zend_string_release_ex(result, 0);
 			RETVAL_FALSE;
 		}
-	} else {
-		RETVAL_FALSE;
 	}
-	EVP_PKEY_CTX_free(ctx);
+
+cleanup:
+	if (pkey) {
```

Comment:
we should probably get rid of all those NULL checks as they are useless - It's already checked by OpenSSL in all supported version (just checked 1.0.2 and master) and we also already have few places where we don't check...

Reply:
Would you like me to do this in this PR or separately? Probably this can be done for all free calls in OpenSSL, not just the EVP_PKEY_free ones.

Reply:
Think there are some exceptions but you are right there more such calls so it probably makes more sense to do later.

The PR looks good so feel free to merge.